src/main/java/seedu/address/model/group/GroupName.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.group;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class GroupName {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String name;
+
+    /**
+     * Constructs a {@code Name}.
+     *
+     * @param name A valid name.
+     */
+    public GroupName(String name) {
+        requireNonNull(name);
+        checkArgument(isValidName(name), MESSAGE_CONSTRAINTS);
+        this.name = name;
+    }
+
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    public static boolean isValidName(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+
+    @Override
+    public String toString() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof GroupName // instanceof handles nulls
+                && name.equals(((GroupName) other).name)); // state check
```

Comment:
I can't really follow this logic. Perhaps we can break it down into smaller parts

Reply:
@marcuspeh The GroupName and UniqueGroupList portion was created by @ramapriyan912001. Perhaps he can modify these parts and make a PR to update the master branch accordingly.

Reply:
@marcuspeh what is it about this that is unclear?

Reply:
What I meant you can split `(other instanceof GroupName && name.equals(((GroupName) other).name))` into a variable first before using that variable in return. Something like what's shown in the screenshot.
![image](https://user-images.githubusercontent.com/54053213/135785013-9de222a1-a4a5-418d-80c1-5268aa31fc47.png)

Reply:
I see - I followed the template provided by ab3's pre-existing `UniquePersonList.java` class

```
    @Override
    public boolean equals(Object other) {
        return other == this // short circuit if same object
                || (other instanceof UniquePersonList // instanceof handles nulls
                        && internalList.equals(((UniquePersonList) other).internalList));
    }
```

Do you feel we should stick to the template? Or try to simplify?

Reply:
I think for now we stick on first. We can do a code quality check after we are done with v1.2 and change all of these